AssertionError: output is not in graph

**Describe the bug**

  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/graph_util_impl.py", line 133, in _assert_nodes_are_present
    assert d in name_to_node, "%s is not in graph" % d
AssertionError: output is not in graph


**Urgency**
High

**System information**
- OS Platform and Distribution: Red Hat Enterprise Linux Server 7.6 (Maipo)
- Tensorflow Version: 1.12
- Python version: 3.5

**To Reproduce**
python -m tf2onnx.convert --input $frozen --inputs $input_names --outputs $output_names --output $output $@

Using this command to generate the onnx model. the $frozen variable is the path to the frozen .pb tf model. 

Tried the same using --checkpoint also, and pointing to path of .meta file and still receiving the same error.


**Expected behavior**
Expect model.onnx to be saved to output location

**Screenshots**
Entire Output of command:

$ python -m tf2onnx.convert --input $frozen --inputs $input_names --outputs $output_names --output $output $@

/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/dtypes.py:523: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/dtypes.py:524: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/dtypes.py:525: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/dtypes.py:526: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/dtypes.py:527: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/dtypes.py:532: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])
Traceback (most recent call last):
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/lib/python3.5/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tf2onnx/convert.py", line 161, in <module>
    main()
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tf2onnx/convert.py", line 116, in main
    graph_def, inputs, outputs = loader.from_graphdef(args.graphdef, args.inputs, args.outputs)
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tf2onnx/loader.py", line 64, in from_graphdef
    frozen_graph = freeze_session(sess, output_names=output_names)
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tf2onnx/loader.py", line 37, in freeze_session
    output_names, freeze_var_names)
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/graph_util_impl.py", line 232, in convert_variables_to_constants
    inference_graph = extract_sub_graph(input_graph_def, output_node_names)
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/graph_util_impl.py", line 174, in extract_sub_graph
    _assert_nodes_are_present(name_to_node, dest_nodes)
  File "/cluster/home/pandyar1/Python3.5/Python-3.5.6/envTensorflow3.5/lib/python3.5/site-packages/tensorflow/python/framework/graph_util_impl.py", line 133, in _assert_nodes_are_present
    assert d in name_to_node, "%s is not in graph" % d
AssertionError: output is not in graph


**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

--- a/tf2onnx/graph.py
+++ b/tf2onnx/graph.py
@@ -1079,7 +1079,7 @@ class Graph(object):
         if name is None:
             name = utils.make_name(node.name)
         new_output = port_name(name)
-        if type(input_name) is not list:
+        if not isinstance(input_name, list):
             input_name = [input_name]
 
         new_node = self.make_node(op_type, input_name, attr=kwargs, outputs=[new_output], name=name, domain=domain)

--- a/tf2onnx/graph.py
+++ b/tf2onnx/graph.py
@@ -1067,7 +1067,9 @@ class Graph(object):
         Args:
             node: we want to replace the input for this node
             op_type: type for new operation
-            input_name: the names of the outputs above us
+            input_name: the name(s) of the outputs above us
+                if scalar, new node placed above input_name
+                if list, new node placed above input_name[0]. list is inputs into new node
             name: the name of the new op
             kwargs: attributes of the new node
 
@@ -1077,9 +1079,12 @@ class Graph(object):
         if name is None:
             name = utils.make_name(node.name)
         new_output = port_name(name)
-        new_node = self.make_node(op_type, [input_name], attr=kwargs, outputs=[new_output], name=name, domain=domain)
+        if type(input_name) is not list:
+            input_name = [input_name]
+
+        new_node = self.make_node(op_type, input_name, attr=kwargs, outputs=[new_output], name=name, domain=domain)
         for i, n in enumerate(node.input):
-            if n == input_name:
+            if n == input_name[0]:
                 node.input[i] = new_output
                 break
         return new_node

--- a/tf2onnx/onnx_opset/generator.py
+++ b/tf2onnx/onnx_opset/generator.py
@@ -16,8 +16,8 @@ from onnx import onnx_pb, numpy_helper
 from tf2onnx import utils
 from tf2onnx.handler import tf_op
 
-
 logger = logging.getLogger(__name__)
+
 
 # pylint: disable=unused-argument,missing-docstring
 
@@ -101,6 +101,15 @@ class Fill:
         value_proto = numpy_helper.from_array(value)
         node.set_attr("value", value_proto)
         del node.input[1]
+
+    @classmethod
+    def version_11(cls, ctx, node, **kwargs):
+        # cls.version_7(ctx, node, **kwargs)
+        node.type = "Expand"
+        node.input = [node.input[1], node.input[0]]
+        # cast shape to int64 if needed
+        if ctx.get_dtype(node.input[1]) != onnx_pb.TensorProto.INT64:
+            ctx.insert_new_node_on_input(node, "Cast", node.input[1], to=onnx_pb.TensorProto.INT64)
 
 
 @tf_op("Multinomial")

--- a/tf2onnx/onnx_opset/nn.py
+++ b/tf2onnx/onnx_opset/nn.py
@@ -286,7 +286,7 @@ class DepthwiseConv2d:
         k_h, k_w, k_input_channels, k_channel_multiplier = kernel_shape
         if k_input_channels < 1:
             raise ValueError("input channel must be positive")
-        k_output_channels = k_input_channels  * k_channel_multiplier
+        k_output_channels = k_input_channels * k_channel_multiplier
 
         node.set_attr("kernel_shape", [k_h, k_w])
         strides = conv_dims_attr(node, "strides")
@@ -448,13 +448,17 @@ class Pad:
         if mode not in [None, "constant", "reflect"]:
             raise ValueError(mode + " pad mode is not supported")
 
-        pads = node.inputs[1].get_tensor_value()
-        pads = np.array(pads).transpose().flatten().astype(np.int64)
-        node.inputs[1].set_tensor_value(pads)
+        # pads must be int64
+        if ctx.get_dtype(node.input[1]) != onnx_pb.TensorProto.INT64:
+            ctx.insert_new_node_on_input(node, "Cast", node.input[1], to=onnx_pb.TensorProto.INT64)
+        ctx.insert_new_node_on_input(node, "Transpose", node.input[1])
+        reshape = ctx.insert_new_node_on_input(node, "Reshape", node.input[1])
+        shape_const = ctx.make_const(utils.make_name(node.name), np.array([-1]).astype(np.int64))
+        reshape.input = [reshape.input[0], shape_const.name]
 
         origin_dtype = ctx.get_dtype(node.output[0])
-        if origin_dtype not in [TensorProto.FLOAT16, TensorProto.FLOAT,
-                                TensorProto.DOUBLE]:
+        if origin_dtype not in [TensorProto.FLOAT, TensorProto.DOUBLE,
+                                TensorProto.INT32, TensorProto.INT64]:
             cast_node = ctx.insert_new_node_on_input(node, "Cast", node.input[0])
             cast_node.set_attr("to", TensorProto.FLOAT)
             ctx.set_dtype(cast_node.output[0], TensorProto.FLOAT)

--- a/tf2onnx/onnx_opset/nn.py
+++ b/tf2onnx/onnx_opset/nn.py
@@ -452,9 +452,8 @@ class Pad:
         if ctx.get_dtype(node.input[1]) != onnx_pb.TensorProto.INT64:
             ctx.insert_new_node_on_input(node, "Cast", node.input[1], to=onnx_pb.TensorProto.INT64)
         ctx.insert_new_node_on_input(node, "Transpose", node.input[1])
-        reshape = ctx.insert_new_node_on_input(node, "Reshape", node.input[1])
         shape_const = ctx.make_const(utils.make_name(node.name), np.array([-1]).astype(np.int64))
-        reshape.input = [reshape.input[0], shape_const.name]
+        ctx.insert_new_node_on_input(node, "Reshape", [node.input[1], shape_const.name])
 
         origin_dtype = ctx.get_dtype(node.output[0])
         if origin_dtype not in [TensorProto.FLOAT, TensorProto.DOUBLE,
@@ -715,8 +714,12 @@ class MatrixBandPart:
         # 2: "loop" to generate mask matrix: generate col or row of matrix one by one
         g = ctx.create_new_graph_with_same_config()
         node_name = utils.make_name("const_zero_bool")
-        const_zero_bool = ctx.make_const(name=node_name, np_val=np.array([[0]]).astype(np.bool))
-        ctx.set_dtype(const_zero_bool.output[0], onnx_pb.TensorProto.BOOL)
+        const_zero_bool = g.make_const(name=node_name, np_val=np.array([[0]]).astype(np.bool))
+        g.set_dtype(const_zero_bool.output[0], onnx_pb.TensorProto.BOOL)
+
+        g.add_graph_input("trip", onnx_pb.TensorProto.INT64, [])
+        g.add_graph_input("cond", onnx_pb.TensorProto.BOOL, [])
+        g.add_graph_input("line", onnx_pb.TensorProto.BOOL, [-1, -1])
 
         # shift right the line and add zero at the left.
         new_line = g.make_node(op_type="Concat", inputs=[const_zero_bool.output[0], "line"],
@@ -729,10 +732,6 @@ class MatrixBandPart:
         g.make_node("Identity", ["cond"], outputs=["cond_out"])
         g.make_node("Identity", ["line"], outputs=["res"])
         g.make_node("Identity", [slice_node], outputs=["line_out"])
-
-        g.add_graph_input("trip", onnx_pb.TensorProto.INT64, [])
-        g.add_graph_input("cond", onnx_pb.TensorProto.BOOL, [])
-        g.add_graph_input("line", onnx_pb.TensorProto.BOOL, [-1, -1])
 
         g.add_graph_output("cond_out", onnx_pb.TensorProto.BOOL, [])
         g.add_graph_output("line_out", onnx_pb.TensorProto.BOOL, [-1, -1])

